pkg/core/interop/context.go

```diff
@@ -101,36 +107,45 @@ type Contract interface {
 type ContractMD struct {
 	state.NativeContract
 	Name    string
-	Methods map[MethodAndArgCount]MethodAndPrice
-}
-
-// MethodAndArgCount represents method's signature.
-type MethodAndArgCount struct {
-	Name     string
-	ArgCount int
+	Methods []MethodAndPrice
 }
 
 // NewContractMD returns Contract with the specified list of methods.
 func NewContractMD(name string, id int32) *ContractMD {
-	c := &ContractMD{
-		Name:    name,
-		Methods: make(map[MethodAndArgCount]MethodAndPrice),
-	}
+	c := &ContractMD{Name: name}
 
 	c.ID = id
 
 	// NEF is now stored in contract state and affects state dump.
 	// Therefore values are taken from C# node.
 	c.NEF.Header.Compiler = "neo-core-v3.0"
 	c.NEF.Header.Magic = nef.Magic
-	c.NEF.Script = state.CreateNativeContractScript(id)
-	c.NEF.Checksum = c.NEF.CalculateChecksum()
-	c.Hash = state.CreateContractHash(util.Uint160{}, c.NEF.Checksum, name)
+	c.Hash = state.CreateContractHash(util.Uint160{}, 0, c.Name)
 	c.Manifest = *manifest.DefaultManifest(name)
 
 	return c
 }
 
+// UpdateHash creates native contract script and updates hash.
+func (c *ContractMD) UpdateHash() {
+	w := io.NewBufBinWriter()
+	for i := range c.Methods {
+		offset := w.Len()
+		c.Methods[i].MD.Offset = offset
+		c.Manifest.ABI.Methods[i].Offset = offset
+		emit.Int(w.BinWriter, 0)
```

Comment:
We need to adjust notary verification price calculator (`(c *Client) CalculateNotaryFee`) regarding to these changes.

Previously verification price was calculated as:
```
int64((nKeys+1))*transaction.NotaryServiceFeePerKey + // fee for NotaryAssisted attribute
fee.Opcode(baseExecFee, // Notary node witness
opcode.PUSHDATA1, opcode.RET, // invocation script
opcode.PUSHINT8, opcode.SYSCALL, opcode.RET) + // System.Contract.CallNative
native.NotaryVerificationPrice + // Notary witness verification price
feePerByte*int64(io.GetVarSize(make([]byte, 66))) + // invocation script per-byte fee
feePerByte*int64(io.GetVarSize([]byte{})), // verification script per-byte fee
```

And now it should be calculated as:
```
int64((nKeys+1))*transaction.NotaryServiceFeePerKey + // fee for NotaryAssisted attribute
fee.Opcode(baseExecFee, // Notary node witness
opcode.PUSHDATA1, opcode.RET, // invocation script
opcode.PUSH0, opcode.SYSCALL, opcode.RET) + // System.Contract.CallNative
native.NotaryVerificationPrice + // Notary witness verification price
feePerByte*int64(io.GetVarSize(make([]byte, 66))) + // invocation script per-byte fee
feePerByte*int64(io.GetVarSize([]byte{})), // verification script per-byte fee
```

So the only difference in PUSHDATA1 -> PUSH0. We have a test for `CalculateNotaryFee` (the test is located in `pkg/rpc/server/client_test.go`, `TestSignAndPushP2PNotaryRequest_good`, section `// it shouldn't be an error to add completed fallback to the chain`), but this test only checks that resulting network fee is enough for transaction acceptance and does not contains opcodes checks. This test is passed because PUSHDATA1 and PUSH0 have the same prices. However, I'd prefer to keep things cleen and adjust `CalculateNotaryFee`.

We also have a test in `core` package with the same problem. See `TestVerifyTx`, subtest `Partially-filled transaction`, function `getPartiallyFilledTx`.